packages/shared-ux/button_toolbar/src/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { ReactElement } from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { IconButtonGroup, PrimaryButton } from '../buttons';
+import { ToolbarPopover } from '../popover';
+
+interface NamedSlots {
+  primaryButton: ReactElement<typeof PrimaryButton | typeof ToolbarPopover>;
+  iconButtonGroup?: ReactElement<typeof IconButtonGroup>;
+  extraButtons?: Array<ReactElement<typeof PrimaryButton | typeof ToolbarPopover> | undefined>;
+}
+
+export interface Props {
+  children: NamedSlots;
+}
+
+export const Toolbar = ({ children }: Props) => {
+  const { primaryButton, iconButtonGroup, extraButtons = [] } = children;
+
+  const extra = extraButtons.map((button, index) =>
```

Comment:
nit: we don't currently restrict this array to any particular length, but it stands to reason there's a maximum.  We might want to consider logging a warning-- or throwing an error-- if the number of extra buttons exceeds a certain design limit.  Thoughts @cchaos ?

Reply:
I added an error in fa629f2 for over 120 buttons. I'm open to lowering that number to...50? Would logging a warning be better? 😅

Reply:
I would log a warning, yes... and I was thinking more like **5**, maybe as many as **7**.  Maybe @ryankeairns can weigh in, but we don't need to block this PR from going in.